### DISCUSSION TITLE
force-cube error 5: expects 3 bands despite 2-band input

### DISCUSSION THREAD

**Discussion by pleitebier** [davidfrantz/force#254] [Q&A] [answered]
Hi again,

after adapting my S1 input to all requirements (Int16, Nodata: -9999, two bands: VV, VH), I get a new error. I had 3-band stacks with VV,VH,VVVH before and reduced the images by the ratio band. Although gdal .rastercount() retrieves only two bands, somehow it seems as if force-cube expects still 3 bands. Any idea how to solve that, would be very welcomed! Thanks in advance.

> ERROR 5: /home/beerf56/brazil/level2/s-america/X0056_Y0029/re_S1A_IW_GRDH_1SDV_20210527T084412_20210527T084437_038071_047E44_A989'_TEMP2.tif': GDALDataset::GetRasterBand(3) - Illegal band #
> 
> Traceback (most recent call last):
>   File "/usr/local/bin/gdal_merge.py", line 611, in <module>
>     sys.exit(main())
>   File "/usr/local/bin/gdal_merge.py", line 596, in main
>     fi.copy_into(t_fh, band, band, nodata)
>   File "/usr/local/bin/gdal_merge.py", line 341, in copy_into
>     return raster_copy(s_fh, sw_xoff, sw_yoff, sw_xsize, sw_ysize, s_band,
>   File "/usr/local/bin/gdal_merge.py", line 107, in raster_copy
>     return raster_copy_with_nodata(
>   File "/usr/local/bin/gdal_merge.py", line 150, in raster_copy_with_nodata
>     data_src = s_band.ReadAsArray(s_xoff, s_yoff, s_xsize, s_ysize,
> AttributeError: 'NoneType' object has no attribute 'ReadAsArray'
> force-cube: merging failed.`

**Comment by pleitebier** [comment]
falls es hilft hier der komplette gdalinfo Auswurf:

> Driver: GTiff/GeoTIFF
> Files: /home/beerf56/brazil/Sentinel1/trial.tif
> Size is 16210, 11001
> Coordinate System is:
> PROJCRS["WGS 84 / UTM zone 23S",
>     BASEGEOGCRS["WGS 84",
>         DATUM["World Geodetic System 1984",
>             ELLIPSOID["WGS 84",6378137,298.257223563,
>                 LENGTHUNIT["metre",1]]],
>         PRIMEM["Greenwich",0,
>             ANGLEUNIT["degree",0.0174532925199433]],
>         ID["EPSG",4326]],
>     CONVERSION["UTM zone 23S",
>         METHOD["Transverse Mercator",
>             ID["EPSG",9807]],
>         PARAMETER["Latitude of natural origin",0,
>             ANGLEUNIT["degree",0.0174532925199433],
>             ID["EPSG",8801]],
>         PARAMETER["Longitude of natural origin",-45,
>             ANGLEUNIT["degree",0.0174532925199433],
>             ID["EPSG",8802]],
>         PARAMETER["Scale factor at natural origin",0.9996,
>             SCALEUNIT["unity",1],
>             ID["EPSG",8805]],
>         PARAMETER["False easting",500000,
>             LENGTHUNIT["metre",1],
>             ID["EPSG",8806]],
>         PARAMETER["False northing",10000000,
>             LENGTHUNIT["metre",1],
>             ID["EPSG",8807]]],
>     CS[Cartesian,2],
>         AXIS["(E)",east,
>             ORDER[1],
>             LENGTHUNIT["metre",1]],
>         AXIS["(N)",north,
>             ORDER[2],
>             LENGTHUNIT["metre",1]],
>     USAGE[
>         SCOPE["Engineering survey, topographic mapping."],
>         AREA["Between 48°W and 42°W, southern hemisphere between 80°S and equator, onshore and offshore. Brazil."],
>         BBOX[-80,-48,0,-42]],
>     ID["EPSG",32723]]
> Data axis to CRS axis mapping: 1,2
> Origin = (252546.937861723650713,8852291.368380291387439)
> Pixel Size = (10.000000000000000,-10.000000000000000)
> Metadata:
>   AREA_OR_POINT=Area
> Image Structure Metadata:
>   INTERLEAVE=PIXEL
> Corner Coordinates:
> Upper Left  (  252546.938, 8852291.368) ( 47d15'35.86"W, 10d22'28.54"S)
> Lower Left  (  252546.938, 8742281.368) ( 47d16' 2.86"W, 11d22' 7.51"S)
> Upper Right (  414646.938, 8852291.368) ( 45d46'46.94"W, 10d22'53.76"S)
> Lower Right (  414646.938, 8742281.368) ( 45d46'56.26"W, 11d22'35.20"S)
> Center      (  333596.938, 8797286.368) ( 46d31'20.74"W, 10d52'34.50"S)
> Band 1 Block=16210x1 Type=Int16, ColorInterp=Gray
>   Description = VV
>   NoData Value=-9999
> Band 2 Block=16210x1 Type=Int16, ColorInterp=Undefined
>   Description = VH
>   NoData Value=-9999

**Comment by davidfrantz** [comment, marked as answer]
Hi @pleitebier,

force-cube only uses GDAL in this case. As you see from the error stack, there is a problem In gdal_merge.py, which is responsible for merging newly cubed files with files that already exist. Is there a possibility that the output directory is not clean and still contains remnants of previous tries?

Cheers,
David

**Reply by pleitebier** [reply]
Indeed...

thanks for the quick help. It's working after cleaning the output directory.

Cheers,
Felix
